src/main/java/bio/terra/workspace/service/workspace/flight/PullProjectFromPoolStep.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.GOOGLE_PROJECT_ID;
+
+import bio.terra.buffer.model.HandoutRequestBody;
+import bio.terra.buffer.model.PoolInfo;
+import bio.terra.buffer.model.ResourceInfo;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.service.buffer.BufferService;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PullProjectFromPoolStep implements Step {
+
+  private final BufferService bufferService;
+
+  private final Logger logger = LoggerFactory.getLogger(PullProjectFromPoolStep.class);
+
+  public PullProjectFromPoolStep(BufferService bufferService) {
+    this.bufferService = bufferService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) {
+    PoolInfo poolInfo = bufferService.getPoolInfo();
```

Comment:
It will return 404 if no resource is available. I'd suggest we check NOT_FOUND error somewhere, and let it return 
STEP_RESULT_FAILURE_RETRY. Then we can config RetryRule to let ask RBS later to see if the pool is refilled.

Reply:
Good point. Changed so it cannot, but is there an annotation to that effect that we use in this code base?

Reply:
It cannot return null. Is there an annotation used in this code base to guarantee that contract?

Reply:
@yonghaoy is that from getPoolInfo or handout resource? If the pool is not found, retrying won't help, right? For handout resource, does NOT_FOUND mean the pool is refilling or the pool is not found?

Reply:
We don't have any annotation for guaranteeing that afaik. Generally we've tried to avoid returning nulls and just documented the methods that can